src/core/storage/utils/state-helpers.ts

```diff
@@ -0,0 +1,442 @@
+import { ExtensionContext } from "vscode"
+import { ApiProvider, BedrockModelId, ModelInfo } from "@shared/api"
+import { LanguageModelChatSelector } from "vscode"
+import { ClineRulesToggles } from "@/shared/cline-rules"
+import { DEFAULT_MCP_DISPLAY_MODE, McpDisplayMode } from "@/shared/McpDisplayMode"
+import { TelemetrySetting } from "@/shared/TelemetrySetting"
+import { UserInfo } from "@/shared/UserInfo"
+import { BrowserSettings, DEFAULT_BROWSER_SETTINGS } from "@/shared/BrowserSettings"
+import { HistoryItem } from "@/shared/HistoryItem"
+import { AutoApprovalSettings, DEFAULT_AUTO_APPROVAL_SETTINGS } from "@/shared/AutoApprovalSettings"
+import { Mode, OpenaiReasoningEffort } from "@/shared/storage/types"
+import { SecretKey } from "../state-keys"
+import { Controller } from "@/core/controller"
+
+export async function readStateFromDisk(context: ExtensionContext) {
```

Comment:
nit: Adding a doc string to explain when this should be used v.s. cacheService would be very helpful here.